### PR TITLE
Issue 350

### LINKED ISSUE #350: Add `get_anomalies_mad` function
### 🚀 Feature Request

Add a function for point anomaly detection that is based on rolling MAD statistics.
Provide the additional possibility of using STL decomposition for optional trend and seasonality removal.

### Proposal

Function signature
```python
def get_anomalies_mad(
    ts: TSDataset,
    in_column: str = "target",
    window_size: int = 10,
    stride: int = 1,
    mad_scale: float = 3,
    trend: bool = False,
    seasonality: bool = False,
    period: Optional[int] = None,
    stl_params: Optional[Dict[str, Any]] = None
) -> Dict[str, Union[List[pd.Timestamp], List[int], pd.Series]]:
    pass
```
`ts` - dataset with series
`in_column` - in which column to perform anomaly detection
`window_size` - size of the window to use for detection
`stride` - how many observations to skip between adjacent windows
`mad_scale` - scaling paremeter for the estimated interval
`trend` - whether to remove a trend from a series
`seasonality` - whether to remove seasonality from series
`period` - periodicity of the sequence for STL
`stl_params` - additional parameters for STL, see here

1. Add a function to the `etna.analysis.outliers` module.
2. Remove leading `NaN`s from the series.
3. If there are some missing values after the previous step, raise an appropriate error.
4. Use STL from [`statsmodels`](https://www.statsmodels.org/devel/generated/statsmodels.tsa.seasonal.STL.html) if parameters are set
   4.1. Subtract trend if the parameter set
   4.2. Subtract seasonality if the parameter set
5. Estimate [MAD](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.median_abs_deviation.html) for each window.
6. Set an observation as anomalous if it falls from the interval at least in one window.
7. Return detected anomalies in a similar format to the other anomaly detection methods.
8. Update module documentation.

### Test cases

* Add tests for the edge cases
* Add tests that passed parameters values are correct
* Add tests similar to other anomaly detection methods

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Polzovat123**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Closing issues
closes #350

### CODE REVIEW

File: etna/analysis/outliers/rolling_statistics.py [hunk 1]
```diff
@@ -275,3 +276,118 @@ def get_anomalies_iqr(
...
+    mad_scale:
+        scaling parameter of the estimated interval.
+    period:
+        periodicity of the sequence for STL.
+    trend:
+        whether to remove trend from the series.
+    seasonality:
+        whether to remove seasonality from the series
+    stl_params:
+        other parameters for STL. See https://www.statsmodels.org/devel/generated/statsmodels.tsa.seasonal.STL.html
```
Comment (brsnw250) [inline etna/analysis/outliers/rolling_statistics.py:308]: Let's use :py:class:`statsmodels.tsa.seasonal.STL` instead of the link.

File: etna/analysis/outliers/rolling_statistics.py [hunk 2]
```diff
@@ -275,3 +276,118 @@ def get_anomalies_iqr(
...
+    if trend or seasonality:
+        if stl_params is None:
+            stl_params = {}
+
+        window = _stl_decompose(series=window, period=period, trend=trend, seasonality=seasonality, **stl_params)
+
+    mad = median_abs_deviation(window)
+    median = np.median(window)
+
+    anom_mask = (window > median + mad_scale * mad) | (window < median - mad_scale * mad)
```
Comment (brsnw250) [inline etna/analysis/outliers/rolling_statistics.py:328]:
```suggestion
    anom_mask = np.abs(window - median) >  mad * mad_scale
```
Let's speed up things a bit here.

File: etna/analysis/outliers/rolling_statistics.py [hunk 3]
```diff
@@ -275,3 +276,118 @@ def get_anomalies_iqr(
...
+    mad_scale: float = 3,
+    trend: bool = False,
+    seasonality: bool = False,
+    period: Optional[int] = None,
+    stl_params: Optional[Dict[str, Any]] = None,
+    index_only: bool = True,
+) -> Dict[str, Union[List[pd.Timestamp], List[int], pd.Series]]:
+    """
+    Get point outliers in time series using median absolute deviation model (estimation model-based method).
+
```
Comment (brsnw250) [inline etna/analysis/outliers/rolling_statistics.py:347]:
```suggestion
    Get point outliers in time series using median absolute deviation.
```

File: etna/analysis/outliers/rolling_statistics.py [hunk 4]
```diff
@@ -275,3 +276,118 @@ def get_anomalies_iqr(
...
+    Outliers are all points deviating from the median by more than mad_scale * median absolute deviation,
+    where median absolute deviation is the median of the absolute deviations from the median in the window.
```
Comment (brsnw250) [inline etna/analysis/outliers/rolling_statistics.py:349]: We don't need to describe in details here, just  mention intervals bounds.

File: tests/test_analysis/test_outliers/test_median_abs_deviation_outliers.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        window_size=window_size,
+        stride=stride,
+        mad_scale=mad_scale,
+        period=period,
+        trend=True,
+    )
+    assert res == right_anomal
+
+
+@pytest.mark.parametrize("true_params", (["1", "2"],))
```
Comment (brsnw250) [inline tests/test_analysis/test_outliers/test_median_abs_deviation_outliers.py:153]: Let's add one more test here, similar to `test_iqr_outliers_full_stl`

**Review by brsnw250 — APPROVED** [review]
